minimum_req.py

```diff
@@ -0,0 +1,159 @@
+    print("Installing the minimum requirements generated")
+    process = ['pip', 'install']
+    process.extend(min_reqs.split(delim)[:-1])
+    # process = [x for x in process if ('wheel' not in x and 'pip==' not in x)]
+    process = [x for x in process if ('pip==' not in x)]
+    subprocess.run(process, capture_output=False)
+    print("Done!")
+
+def get_min_test_and_core_requirements(core_requirements: str, test_requirements: str) -> tuple:
+    # input arguments should be the path to the core and test requirements.txt files
+    min_reqs = ''
+    min_core_reqs, min_test_reqs = [], []
+    with open(test_requirements, "r") as f:
+        min_reqs = f.read()
+        min_test_reqs = min_reqs.split("\n")[:-1]
+
+    with open(core_requirements, "r") as f:
+        s = f.read()
+        min_reqs += s
+        min_core_reqs = s.split("\n")[:-1]
+
+    return (min_reqs, min_core_reqs, min_test_reqs)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Get min dependencies of min dependencies")
+    parser.add_argument('--package', default='woodwork', required=False)
+    parser.add_argument('--json-filename', default='mindep.json', required=False)
+    parser.add_argument('--write-txt', default="False", required=False)
+    parser.add_argument('--output-name', default='min_min_dep.txt', required=False)
+    parser.add_argument('--delimiter', default='\n', required=False)
+    parser.add_argument('--install', default="True", required=False)
+    parser.add_argument('--req-file-path', default='woodwork/tests/requirement_files/', required=False)
+    args = parser.parse_args()
+
+    # get the arguments from the parser
+    package = args.package
+    json_name = args.json_filename
+    delim = args.delimiter
+    write = bool(args.write_txt=='True')
+    output_name = args.output_name
+    install = bool(args.install=='True')
+    path = args.req_file_path
+    
+    # get the minimum requirements and install the min core requirements
+    min_reqs, min_core_reqs, min_test_reqs = get_min_test_and_core_requirements(path + "minimum_core_requirements.txt", path + "minimum_test_requirements.txt")
+    install_min_deps()
+
+    package_to_version_dict = {}
+    all_requirements = []
+
+    # find all packages that the core requirements rely on
+    for package in min_core_reqs:
+        pack = tuple(requirements.parse(package))[0]
+        _package_name = pack.name
+        _package = pkg_resources.working_set.by_key[_package_name]
+        all_requirements.append(package)
+        reliance = [str(r) for r in _package.requires()]
+        all_requirements.extend(reliance)  # retrieve deps from setup.py
+
+    # for any requirements in our test-requirements that apply to the core requirement packages, add that in as well
+    for package in min_test_reqs:
+        pack = tuple(requirements.parse(package))[0]
+        _package_name = pack.name
+        if any([_package_name in s for s in all_requirements]):
```

Comment:
Why do we use `_package_name in s` as opposed to `_package_name == s`? Also just so I understand, why do we append `all_requirements` with a package it already has?